### PR TITLE
Use moveit_configs_utils for launch files

### PR DESCRIPTION
**Pull request by JafarAbdi**
NOTE: you need https://github.com/ros-planning/moveit2/pull/1274 and https://github.com/PickNikRobotics/rosparam_shortcuts/pull/27 to be able to run the demos

### CODE REVIEW

File: demo/launch/alternative_path_costs.launch.py [hunk 1]
```diff
@@ -1,72 +1,28 @@
 import os
```
Comment (stephanie-eng) [inline demo/launch/alternative_path_costs.launch.py:1]:
```suggestion

```
Should be able to remove this now that we don't need to `os.path.join` in `load_file` and `load_yaml` anymore

File: demo/launch/alternative_path_costs.launch.py [hunk 2]
```diff
@@ -1,72 +1,28 @@
...
-import yaml
+
+from ament_index_python.packages import get_package_share_directory
```
Comment (stephanie-eng) [inline demo/launch/alternative_path_costs.launch.py:3]: Do we need `get_package_share_directory`? I don't see it used in this file.

File: demo/launch/cartesian.launch.py [hunk 3]
```diff
@@ -1,55 +1,27 @@
 import os
```
Comment (stephanie-eng) [inline demo/launch/cartesian.launch.py:1]:
```suggestion

```

File: demo/launch/demo.launch.py [hunk 4]
```diff
@@ -1,91 +1,21 @@
 import os
-import yaml
+
+import xacro
```
Comment (stephanie-eng) [inline demo/launch/demo.launch.py:3]:
```suggestion

```
Do we need xacro here? Don't see any use of it in this file.

File: demo/launch/ik_clearance_cost.launch.py [hunk 5]
```diff
@@ -1,55 +1,27 @@
 import os
```
Comment (stephanie-eng) [inline demo/launch/ik_clearance_cost.launch.py:1]:
```suggestion

```

File: demo/launch/modular.launch.py [hunk 6]
```diff
@@ -1,55 +1,27 @@
 import os
-import yaml
+
+from ament_index_python.packages import get_package_share_directory
```
Comment (stephanie-eng) [inline demo/launch/modular.launch.py:3]:
```suggestion

```

File: demo/launch/modular.launch.py [hunk 7]
```diff
@@ -1,55 +1,27 @@
 import os
```
Comment (stephanie-eng) [inline demo/launch/modular.launch.py:1]:
```suggestion

```

File: core/test/move_to.launch.py [hunk 8]
```diff
@@ -1,84 +1,37 @@
-import unittest
 import os
-import yaml
+import unittest
 
+import launch_testing
+import pytest
+from ament_index_python.packages import get_package_share_directory
```
Comment (stephanie-eng) [inline core/test/move_to.launch.py:6]:
```suggestion

```

File: demo/launch/ik_clearance_cost.launch.py (hunk 5 above)
Comment (stephanie-eng) [reply demo/launch/ik_clearance_cost.launch.py:1]: Also, for what it's worth, I couldn't get this example to work when testing, but I don't see that being related to this PR at all.

**Review by stephanie-eng — COMMENTED** [review]
I found a few imports I think we can clean up

File: demo/launch/ik_clearance_cost.launch.py (hunk 5 above)
Comment (JafarAbdi) [reply demo/launch/ik_clearance_cost.launch.py]: Do you mind sharing the error you're getting?

Comment (stephanie-eng) [reply demo/launch/ik_clearance_cost.launch.py]: I think the issue is actually a problem with rosparam_shortcuts that looks related to your PR https://github.com/PickNikRobotics/rosparam_shortcuts/pull/27

Here's the error:

```
[ik_clearance_cost-1] terminate called after throwing an instance of 'rclcpp::exceptions::UninitializedStaticallyTypedParameterException'
[ik_clearance_cost-1]   what():  Statically typed parameter 'cumulative' must be initialized.
[ERROR] [ik_clearance_cost-1]: process has died [pid 642360, exit code -6, cmd '/workspaces/ros2-rolling-ws/install/moveit_task_constructor_demo/lib/moveit_task_constructor_demo/ik_clearance_cost --ros-args --params-file /tmp/launch_params_etkxjhf0 --params-file /tmp/launch_params_c324op82 --params-file /tmp/launch_params_26xnvx2h'].
```
I also noticed a small typo, please look at #366 if you get a chance!

File: demo/launch/cartesian.launch.py [hunk 9]
```diff
@@ -1,55 +1,24 @@
...
 
     cartesian_task = Node(
         package="moveit_task_constructor_demo",
         executable="cartesian",
         output="screen",
-        parameters=[robot_description, robot_description_semantic, kinematics_yaml],
+        parameters=[
+            moveit_config.robot_description,
+            moveit_config.robot_description_semantic,
+            moveit_config.robot_description_kinematics,
```
Comment (henningkayser) [inline demo/launch/cartesian.launch.py:20]: Shouldn't this one also have planning pipelines?

**Review by henningkayser — APPROVED** [review]
This is such a huge improvement!

File: demo/launch/cartesian.launch.py (hunk 9 above)
Comment (JafarAbdi) [reply demo/launch/cartesian.launch.py:20]: I don't think we need it since we use the `CartesianInterpolator` in this demo

**Pull request merged**